### PR TITLE
Added a unit test to verify switching media groups during playback works

### PR DESCRIPTION
**Pull request by BrandonOCasey**
silenced videojs.log during tests
updated multipleAudioGroups files to be easier to understand
changed caret character to single quote where we were not using it as a template string
Pulls in changes from #642

### CODE REVIEW

File: test/test-helpers.js [hunk 1]
```diff
@@ -112,12 +112,21 @@ let fakeEnvironment = {
...
+    });
+
   }
 };
 
 export const useFakeEnvironment = function() {
   fakeEnvironment.clock = sinon.useFakeTimers();
-
+  ['warn', 'error'].forEach((level) => {
+    sinon.stub(videojs.log, level);
```
Comment (gesinger) [inline test/test-helpers.js]: I agree that the warnings are a pain in the tests, but are they something we should address? Or can we never get around them, as we need to use the internals of videojs?

File: test/videojs-contrib-hls.test.js [hunk 2]
```diff
@@ -1793,6 +1793,81 @@ QUnit.test('cleans up the buffer when loading VOD segments', function() {
...
+  let hls = this.player.tech_.hls;
+  let mpc = hls.masterPlaylistController_;
+  let at = this.player.audioTracks();
+
+  QUnit.equal(at.length, 3, 'three audio tracks after load');
+  let trackOne = at[0];
+  let trackTwo = at[1];
+  let trackThree = at[2];
+
+  QUnit.equal(trackOne.enabled, true, 'track on enabled after load');
```
Comment (gesinger) [inline test/videojs-contrib-hls.test.js]: Minor type (on => one)

Comment (gesinger) [inline test/videojs-contrib-hls.test.js]: I even made a typo in the last comment...

File: test/videojs-contrib-hls.test.js [hunk 3]
```diff
@@ -1793,6 +1793,81 @@ QUnit.test('cleans up the buffer when loading VOD segments', function() {
...
+
+  let oldMediaGroup = mpc.media().attributes.AUDIO;
+
+  // force mpc to select a new media group
+  mpc.hlsHandler.selectPlaylist = () => {
+    let select = null;
+
+    mpc.master().playlists.forEach((playlist) => {
```
Comment (gesinger) [inline test/videojs-contrib-hls.test.js]: We can probably use:

```javascript
return mpc.master().playlists.find(playlist => playlist.attributes.AUDIO !== oldMediaGroup);
```

**Comment by gesinger** [comment]
Nice test addition. Some minor comments, but otherwise, looks good.

File: test/test-helpers.js (hunk 1 above)
Comment (BrandonOCasey) [inline test/test-helpers.js]: I think we may be able to get rid of this issue without silencing the log, took a few minutes to narrow it down, I will write a test around it to verify that nothing changes basically it involves the depreciation warning when we use `player.hls`. when we were checking if `player.hls` existed when it did exist we inadvertently called it and caused the warning to showup.

File: test/videojs-contrib-hls.test.js (hunk 2 above)
Comment (BrandonOCasey) [inline test/videojs-contrib-hls.test.js]: :)

File: test/videojs-contrib-hls.test.js (hunk 3 above)
Comment (imbcmdth) [inline test/videojs-contrib-hls.test.js]: I don't think `find` is polyfilled.. it'll work in Chrome but not everywhere.

**Comment by BrandonOCasey** [comment]
ok so ultimately I decided that it would be good to force us to make sure that we are logging. So I mock the log as a did previously and on this.env.restore() we verify that log functions were called 0 times. I added a getter to the log stubbing that will reset it to zero. Basically if you check how many calls happened with that then we know you checked (I guess you could still throw it to the void but CR's would catch it).

File: test/videojs-contrib-hls.test.js (hunk 3 above)
Comment (BrandonOCasey) [inline test/videojs-contrib-hls.test.js]: seems to work on Firefox/Chrome

**Pull request merged**